crates/subspace-networking/src/node_runner.rs

```diff
@@ -135,9 +154,30 @@ impl NodeRunner {
     }
 
     async fn handle_peer_dialing(&mut self) {
-        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
         let local_peer_id = *self.swarm.local_peer_id();
+        let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
 
+        // Handle reserved peers first.
+        if !self.reserved_peers.is_empty() {
+            trace!(%local_peer_id, "Checking reserved peers connection: {:?}", self.reserved_peers);
+
+            let reserved_peers = HashMap::<PeerId, Multiaddr>::from_iter(
+                convert_multiaddresses(self.reserved_peers.clone()).into_iter(),
+            );
+            let connected_peers_id_set = HashSet::from_iter(connected_peers.clone());
+            let reserved_peers_id_set = reserved_peers.keys().cloned().collect::<HashSet<_>>();
+
+            let missing_reserved_peer_ids =
+                reserved_peers_id_set.difference(&connected_peers_id_set);
+
+            for peer_id in missing_reserved_peer_ids {
+                if let Some(addr) = reserved_peers.get(peer_id) {
+                    self.dial_peer(*peer_id, addr.clone());
+                }
+            }
```

Comment:
Sorry for late comment, but I think it is always better to use `.collect()` in oppose to `from_iter()`. But this is a minor thing
```suggestion
            let reserved_peers = 
                convert_multiaddresses(self.reserved_peers.clone()).into_iter().collect::<HashMap<_, _>>();
            let connected_peers_id_set = connected_peers.iter().cloned().collect();
            let reserved_peers_id_set = reserved_peers.keys().cloned().collect::<HashSet<_>>();

            for peer_id in reserved_peers_id_set.difference(&connected_peers_id_set) {
                if let Some(addr) = reserved_peers.get(peer_id) {
                    self.dial_peer(*peer_id, addr.clone());
                }
            }
```

Reply:
I kept missing_reserved_peer_ids as 'code comment'.